orangecontrib/imageanalytics/widgets/owimageviewer.py

```diff
@@ -223,6 +225,7 @@ class Outputs:
     titleAttr = settings.ContextSetting(0)
 
     imageSize = settings.Setting(100)  # type: int
+    selected_indices: List[int] = settings.Setting([], schema_only=True)
```

Comment:
Wouldn't it be better to store file names, that is, values from `imageAttr` column? If data is changed so that some images are missing, the indices in the stored selection will be off. For instance, I use images from Import Images, save the workflow, closed it, deleted one of the two images, reopened the workflow, and another image was selected instead of the deleted.

Storing indices also depends upon the OS always returning file names in the same order.

Also, selection is now lost at any change of the input signal. What about making this a schema-only context setting?

Reply:
> Wouldn't it be better to store file names, that is, values from imageAttr column? If data is changed so that some images are missing, the indices in the stored selection will be off. For instance, I use images from Import Images, save the workflow, closed it, deleted one of the two images, reopened the workflow, and another image was selected instead of the deleted.

Aggre. I was copying how other widgets in Orange save selection (with indexes), but I agree we can store image names instead. I will propose a change.

> Also, the selection is now lost at any change of the input signal. What about making this a schema-only context setting?

This is how selection works in most of the other Orange widgets (Scatter Plot, Line Plot, ...). I think context setting is not the right solution here since image names (that are the content of the table) do not depend on the domain (totally, another set of images can have the same domain, especially when they are imported with the Import Images widget). If we memorize the names of images, it can actually be a normal schema_only Setting without being reset on new data on input.